src/services/pullRequest.services.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Checks if a specific file is part of the changes in a pull request.
+ *
+ * @param {InstanceType<typeof GitHub>} octokit - An Octokit instance.
+ * @param {string} owner - The repository owner.
+ * @param {string} repo - The repository name.
+ * @param {string} prNumber - The pull request number.
+ * @param {string} path - The file path.
+ * @returns {Promise<object>} - An object containing the file details.
+ * @throws {Error} - If an error occurs while fetching the file.
+ */
+const isFileInCommitedChanges = async (
+  octokit,
+  owner,
+  repo,
+  prNumber,
+  path
+) => {
+  try {
+    let page = 1;
+    let files;
+    do {
+      // Get the list of files for the pull request for the current page
+      ({ data: files } = await octokit.rest.pulls.listFiles({
+        owner,
+        repo,
+        pull_number: prNumber,
+        per_page: 1,
```

Comment:
```suggestion
```

The default is 30, I wouldn't decrease that. If you do this the current way, you are fetching one file per request. If its a big PR this can take forever and also run into rate limitations.

Reply:
My bad! I was testing the code for different scenarios. Your suggestion is the right one!